docs/guides/deploy.md

```diff
@@ -53,97 +58,95 @@ from concrete import fhe
 client = fhe.Client(client_specs)
 ```
 
-## Generating keys (on the client)
+### Generating keys (client-side)
 
-Once you have the `Client` object, you can perform key generation:
+9. **Generate keys**: Once you have the `Client` object, perform key generation. This method generates encryption/decryption keys and evaluation keys. 
 
 <!--pytest-codeblocks:skip-->
 ```python
 client.keys.generate()
 ```
 
-This method generates encryption/decryption keys and evaluation keys.
 
-The server needs access to the evaluation keys that were just generated. You can serialize your evaluation keys as shown:
+10. **Serialize the evaluation keys**: The server needs access to the evaluation keys. You can serialize your evaluation keys as below.
 
 <!--pytest-codeblocks:skip-->
 ```python
 serialized_evaluation_keys: bytes = client.evaluation_keys.serialize()
 ```
-
-After serialization, send the evaluation keys to the server.
+11. **Send the evaluation keys to the server**.
 
 {% hint style="info" %}
-Serialized evaluation keys are very large, so you may want to cache them on the server instead of sending them with each request.
+Serialized evaluation keys are very large. Consider caching them on the server instead of sending them with each request.
 {% endhint %}
 
-## Encrypting inputs (on the client)
+### Encrypting inputs (client-side)
 
-The next step is to encrypt your inputs and request the server to perform some computation. This can be done in the following way:
+12. **Encrypt inputs**: Encrypt your inputs and request the server to perform some computation. This can be done in the following way.
 
 <!--pytest-codeblocks:skip-->
 ```python
 arg: fhe.Value = client.encrypt(7)
 serialized_arg: bytes = arg.serialize()
 ```
 
-Then, send the serialized arguments to the server.
+13. **Send the serialized arguments to the server**.
 
-## Performing computation (on the server)
+### Performing computation (server-side)
 
-Once you have serialized evaluation keys and serialized arguments, you can deserialize them:
+14. **Deserialize received data**: On the server, deserialize the received evaluation keys and arguments received from the client.
 
 <!--pytest-codeblocks:skip-->
 ```python
 deserialized_evaluation_keys = fhe.EvaluationKeys.deserialize(serialized_evaluation_keys)
 deserialized_arg = fhe.Value.deserialize(serialized_arg)
 ```
 
-You can perform the computation, as well:
+15. **Run the computation**: Perform the computation and serialize the result.
 
 <!--pytest-codeblocks:skip-->
 ```python
 result: fhe.Value = server.run(deserialized_arg, evaluation_keys=deserialized_evaluation_keys)
 serialized_result: bytes = result.serialize()
 ```
 
-Then, send the serialized result back to the client. After this, the client can decrypt to receive the result of the computation.
+16. **Send the serialized result to the client**: After this step, the client can decrypt to receive the result of the computation.
```

Comment:
I would remove "After this step, the client can decrypt to receive the result of the computation." since actually, it's explained in further steps, ie 16 [ the second one!] and 17